ipaserver/install/server/upgrade.py

```diff
@@ -594,14 +594,26 @@ def named_add_ipa_ext_conf_file():
         logger.info('DNS is not configured.')
         return False
 
+    # migrate value from named.conf
+    dnssec_validation = bindinstance.named_conf_get_directive(
+        "dnssec-validation",
+        bindinstance.NAMED_SECTION_OPTIONS,
+        str_val=False
+    )
+    if dnssec_validation is None:
+        dnssec_validation = "yes"
+
     tasks = [
         bindinstance.named_add_ext_conf_file(
             paths.NAMED_CUSTOM_CFG_SRC,
             paths.NAMED_CUSTOM_CONFIG
         ),
         bindinstance.named_add_ext_conf_file(
             paths.NAMED_CUSTOM_OPTIONS_CFG_SRC,
-            paths.NAMED_CUSTOM_OPTIONS_CONFIG
+            paths.NAMED_CUSTOM_OPTIONS_CONFIG,
+            dict(
+                NAMED_DNSSEC_VALIDATION=dnssec_validation
+            )
```

Comment:
The upgrade will create the new file in /etc/named/ipa-options-ext.conf but it will leave the existing option dnssec-validation in /etc/named.conf.
I don't know how bind handles the same option defined in 2 different files but it may be confusing for the admin and lead to errors. The code should probably remove the existing option from /etc/named.conf.

Reply:
There is no upgrade path from old to new configuration yet. I'm working on upgrade in another (not yet published) PR.

Reply:
There are also other issues:
- each time the ipa-server-upgrade tool is executed, a line with `include "/etc/named/ipa-ext.conf";` is added to /etc/ipa/named.conf. The sysupgrade state file contains a section for [named.conf] that can be leveraged.
- the line `include "/etc/named/ipa-options-ext.conf";` is added on clean install but not on upgrade

Reply:
Thanks Flo! I didn't know about the first issue.
The second one is to be expected. There is no upgrade code yet.